### PR TITLE
Minor clean up

### PR DESCRIPTION
**Pull request by conr2d**
This PR performs minor code cleanup to reduce verbosity. Since the compiler has already optimized out indirect calls in the existing code, these changes improve readability but do not affect performance.

### CODE REVIEW

File: substrate/primitives/runtime/src/lib.rs [hunk 1]
```diff
@@ -438,34 +438,25 @@ impl TryFrom<MultiSigner> for ecdsa::Public {
...
-			(Self::Sr25519(ref sig), who) => match sr25519::Public::from_slice(who.as_ref()) {
-				Ok(signer) => sig.verify(msg, &signer),
-				Err(()) => false,
-			},
-			(Self::Ecdsa(ref sig), who) => {
+		let who: &[u8; 32] = signer.as_ref();
+		match (self, *who) {
+			(Self::Ed25519(sig), who) => sig.verify(msg, &who.into()),
+			(Self::Sr25519(sig), who) => sig.verify(msg, &who.into()),
+			(Self::Ecdsa(sig), who) => {
```
Comment (bkchr) [inline substrate/primitives/runtime/src/lib.rs:456]:
```suggestion
		match self {
			Self::Ed25519(sig) => sig.verify(msg, &who.into()),
			Self::Sr25519(sig) => sig.verify(msg, &who.into()),
			Self::Ecdsa(sig) => {
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@conr2d please add a `prdoc`.

**Comment by conr2d** [comment]
@bkchr Thank you for your review.

Since `sr25519::Public` and `ed25519::Public` implement `From<[u8; 32]>` but not `From<&[u8; 32]>`, your change would cause a build failure. I’ve updated the code to clone the inner `[u8; 32]` from `AccountId32` explicitly before converting to the public key. This avoids performance loss since `[u8; 32]` will be moved during initialization. I would appreciate it if you could provide feedback on this update.

**Comment by bkchr** [comment]
Okay ty @conr2d, looks good to me

File: prdoc/pr_5284.prdoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+topic: runtime
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Refactor `MultiSignature` and `MultiSigner` code to reduce verbosity.
+
+crates:
+  - name: sp-runtime
+    bump: patch
```
Comment (bkchr) [inline prdoc/pr_5284.prdoc:12]:
```suggestion
    bump: none
```

File: prdoc/pr_5284.prdoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+title: Minor clean up
+author: conr2d
+topic: runtime
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Refactor `MultiSignature` and `MultiSigner` code to reduce verbosity.
```
Comment (bkchr) [inline prdoc/pr_5284.prdoc:8]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Review by davxy — APPROVED** [review]

**Pull request merged**